File: NAP_II_JAVA/ClasseProdutoComboEItem/ProdutoCombo/ProdutoCombo.java
import java.util.List;

public class ProdutoCombo extends Produto {
    private String tipoDesconto;
    private double valorDesconto;
    private List<Produto> listaProdutos; // Lista de produtos que compõem o combo

    // Construtor
    public ProdutoCombo(String nome, double preco, double volume, double peso, String tipoDesconto, double valorDesconto, List<Produto> listaProdutos) {
        super(nome, preco, volume, peso);
        this.tipoDesconto = tipoDesconto;
        this.valorDesconto = valorDesconto;
        this.listaProdutos = listaProdutos;
    }

    // Getters e setters
    public String getTipoDesconto() {
        return tipoDesconto;
    }

    public void setTipoDesconto(String tipoDesconto) {
        this.tipoDesconto = tipoDesconto;
    }

    public double getValorDesconto() {
        return valorDesconto;
    }

    public void setValorDesconto(double valorDesconto) {
        this.valorDesconto = valorDesconto;
    }

    public List<Produto> getListaProdutos() {
        return listaProdutos;
    }

    public void setListaProdutos(List<Produto> listaProdutos) {
        this.listaProdutos = listaProdutos;
    }
}

File: NAP II - POO/Python/UFRAcommerce/produto_combo.py
from .produto import Produto

class ProdutoCombo(Produto):
    def __init__(self, nome, tipo_desconto, valor_desconto, produtos):
        super().__init__(nome, preco=0, volume=0, peso=0)
        self.tipo_desconto = tipo_desconto
        self.valor_desconto = valor_desconto
        self.produtos = produtos

    def getPreco(self):
        total = sum(produto.preco for produto in self.produtos)
        if self.tipo_desconto == "absoluto":
            return max(0, total - self.valor_desconto)
        elif self.tipo_desconto == "percentual":
            return total * (1 - self.valor_desconto / 100)
        return total

    def __str__(self):
        return f"ProdutoCombo(nome={self.nome}, produtos=[{', '.join(p.nome for p in self.produtos)}], preco={self.getPreco()})"
